File: 2022-06-craft-conf-languages/examples/memory.go
package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func ProcessFamily() {
	me := Person{Name: "Doug", Age: 42}
	fmt.Printf("%s is %d years old\n", me.Name, me.Age)
}

func main() {
	ProcessFamily()
}

File: 2022-06-craft-conf-languages/examples/memory.rs
use std::sync::Arc;

struct Person {
    name: Box<String>,
    age: u8,
}

impl Person {
    fn new(name: &str, age: u8) -> Person {
        let name = name.to_string();
        let name = Box::new(name);
        Person { name, age }
    }
}

fn process_family() {
    let me = Arc::new(Person::new("Doug", 42));
    println!("{} is {} years old", me.name, me.age);
}

fn main() {
    process_family();
}
